igcollect/ping.py

```diff
@@ -36,21 +36,26 @@ def parse_args():
 def main():
     args = parse_args()
 
+    chunked_hosts = [args.hosts[i:i + 10]
+                     for i in range(0, len(args.hosts), 10)]
+
+    data = [(args.prefix, chunk, args.count, args.timeout)
+            for chunk in chunked_hosts]
+
     with Pool(cpu_count() * 2) as p:
-        for i in range(len(args.hosts) % 10):
-            p.starmap(ping, [(args.prefix, args.hosts[i*10:i*10+9],
-                              args.count, args.timeout)])
-        p.starmap(ping, [(args.prefix,
-                          args.hosts[(len(args.hosts) % 10) * 10:
-                                     len(args.hosts) - 1],
-                          args.count, args.timeout)])
+        p.starmap(ping, data)
+
 
 
 def ping(prefix, hosts, count, timeout):
     hosts_string = " ".join(hosts)
     # -p calculates the time fping waits between single ping probes
-    cmd = ('fping -B1 -q -C {} -p {} -t {} {}'.format(count, (20/count) * 1800,
-                                                      timeout, hosts_string))
+    # fping used 20 probes/min by default und set this value to 1800
```

Comment:
Still doesn't explain why exactly 1800? What makes it important to wait exactly 1800 milliseconds between each try when passing 20.
And what's the reason we don't provide this as a parameter to just pass in? Maybe I am missing something here, but I don't see the point of having such a fixed calculation here?